File: Metode avansate de Programare/Teme/familydoctor/saptamana 1-7/Java/MedicDeFamilie/src/familydoctor/model/Appointment.java
package familydoctor.model;

public class Appointment {
    private String id, patientId, diagnostic;
    private boolean isEmergency;

    public Appointment() {

    }

    public Appointment(String id, String patientId, String diagnostic, boolean isEmergency) {
        this.id = id;
        this.patientId = patientId;
        this.diagnostic = diagnostic;
        this.isEmergency = isEmergency;

        validate();
    }

    public String getId() {
        return id;
    }
    public void setId(String value) {
        id = value;
    }

    public String getPatientId() {
        return patientId;
    }
    public void setPatientId(String value) {
        patientId = value;
    }

    public String getDiagnostic() {
        return diagnostic;
    }
    public void setDiagnostic(String value) {
        diagnostic= value;
    }

    public boolean getIsEmergency() {
        return isEmergency;
    }
    public void setIsEmergency(boolean value) {
        isEmergency = value;
    }

    public void validate()
    {
        if (id == null || id == "")
            throw new ValidationException("Invalid ID");

        if (patientId == null || patientId == "")
            throw new ValidationException("Invalid name");

        if (diagnostic == null || diagnostic == "")
            throw new ValidationException("Invalid age");
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;

        if (obj == null)
            return false;

        if (getClass() != obj.getClass())
            return false;

        Appointment other = (Appointment) obj;
        if (id == null) {
            if (other.getId() != null)
                return false;
        } else if (!id.equals(other.getId()))
            return false;

        return true;
    }



}

File: Metode avansate de Programare/Teme/familydoctor/saptamana 1-7/CSharp/FamilyDoctor/FamilyDoctor/Domain/Appointment.cs
using System;

namespace FamilyDoctor.Domain
{
    public class Appointment
    {
        private string id, patientId, diagnostic;
        private bool isEmergency;

        public string ID { get { return id; } set { id = value; } }
        public string PatientID { get { return patientId; } set { patientId = value; } }
        public string Diagnostic { get { return diagnostic; } set { diagnostic = value; } }
        public bool IsEmergency { get { return isEmergency; } set { isEmergency = value; } }

        public Appointment()
        {

        }

        public Appointment(string id, string patientId, string diagnostic, bool isEmergency = false)
        {
            this.id = id;
            this.patientId = patientId;
            this.diagnostic = diagnostic;
            this.isEmergency = isEmergency;
        }
        
        public void Validate()
        {
            if (id == null || id == string.Empty)
                throw new ValidationException("Invalid ID");

            if (patientId == null || patientId == string.Empty)
                throw new ValidationException("Invalid name");

            if (diagnostic == null || diagnostic == string.Empty)
                throw new ValidationException("Invalid age");
        }

        public override bool Equals(object other)
        {
            if (other == null)
                return false;

            if (other.GetType() == typeof(Appointment))
                if (id == ((Appointment)other).ID)
                    return true;
            return false;
        }

        public override string ToString()
        {
            return id + " (PatientID:" + patientId + ", Diag:" + diagnostic + ", IsEmergency:" + isEmergency + ")";
        }
    }
}
